x/merkledb/trieview.go

```diff
@@ -162,22 +162,30 @@ func newTrieView(
 
 	for _, op := range changes.BatchOps {
 		newVal := maybe.Nothing[[]byte]()
+		key := op.Key
 		if !op.Delete {
 			val := op.Value
 			if !changes.ConsumeBytes {
 				val = slices.Clone(op.Value)
 			}
 			newVal = maybe.Some(val)
 		}
-		if err := newView.recordValueChange(newPath(op.Key), newVal); err != nil {
+		if !changes.ConsumeBytes {
+			key = slices.Clone(op.Key)
+		}
+		if err := newView.recordValueChange(db.newPath(key), newVal); err != nil {
 			return nil, err
 		}
 	}
 	for key, val := range changes.MapOps {
+		bytesKey := stringToByteSlice(key)
+		if !changes.ConsumeBytes {
+			bytesKey = slices.Clone(bytesKey)
+		}
```

Comment:
Either this doesn't feel needed - or this always feels needed... Either way I feel like something is weird here